doc/api.md

```diff
@@ -765,19 +765,61 @@ following to CB:
 The IoTA executes the transformaion looping over the `attributes` provision field. Every time a new expression is 
 evaluated, the JEXL context is updated with the expression result. The order defined in the `attributes` array is 
 taken for expression evaluation. This should be considered when using **nested expressions**, that uses values 
-calculated in other attributes. For example, for a given provision like the following one:
+calculated in other attributes. 
+
+For example, let's consider the following provision for a device which send a measure named `level`:
+
+```json
+"attributes": [
+    {
+        "name": "correctedLevel",
+        "type": "Number",
+        "expression": "level * 0.897"
+    },
+    {
+        "name": "normalizedLevel",
+        "type": "Number",
+        "expression": "correctedLevel / 100"
+    }
+]
+```
+
+The expression for `correctedLevel` is evaluated first (using `level` measure as input). Next, the `normalizedLevel` is evaluated (using `correctedLevel` calculated attribute, just calculated before).
+
+Note that if we reserve the order, this way:
+
+```json
+"attributes": [
+    {
+        "name": "normalizedLevel",
+        "type": "Number",
+        "expression": "correctedLevel / 100"
+    },
+    {
+        "name": "correctedLevel",
+        "type": "Number",
+        "expression": "level * 0.897"
+    },    
+]
+```
+
+It is not going to work. The first expression expects a `correctedLevel` which is neither a measure (remember the only measure sent by the device is named `level`) nor a previously calculated attribute. Thus, `correctedLevel` will end with a `null` value.
+
+In conclusion: **the order of attributes in the `attributes` arrays at provising time matters with regards to nested expression evaluation**.
+
+Let's consider the following example. It is an anti-pattern but it's quite illustrative on how ordering works:
 
 ```json
 "attributes": [
     {
-        "name": "attrA",
+        "name": "A",
```

Comment:
Better to use `A` and `B` as `attrX` is not correct for measure case (previously used in "When receiving a measure with the following values:").

Btw, I think there was a typo in second provision. It was using name:attrB-expression:attrB (now it uses name:A-exprresion:B)

NTC (just informative)